mcstatus/tests/test_server.py

```diff
@@ -166,30 +166,30 @@ def test_query(self):
             )
         )
 
-        self.socket.remaining = Mock()
-        self.socket.remaining.side_effect = [15, 208]
-
-        with patch("mcstatus.server.UDPSocketConnection") as connection, patch.object(
-            self.server.address, "resolve_ip"
-        ) as resolve_ip:
-            connection.return_value = self.socket
-            resolve_ip.return_value = "127.0.0.1"
-            info = self.server.query()
-
-        conn_bytes = self.socket.flush()
-        assert conn_bytes[:3] == bytearray.fromhex("FEFD09")
-        assert info.raw == {
-            "hostname": "A Minecraft Server",
-            "gametype": "SMP",
-            "game_id": "MINECRAFT",
-            "version": "1.8",
-            "plugins": "",
-            "map": "world",
-            "numplayers": "3",
-            "maxplayers": "20",
-            "hostport": "25565",
-            "hostip": "192.168.56.1",
-        }
+        with patch("mcstatus.protocol.connection.Connection.remaining"):
+            self.socket.remaining.side_effect = [15, 208]
```

Comment:
It's a bit weird to directly access the mocked feature and set the side-effect rather than simply using `as` to obtain that mock.

```suggestion
        with patch("mcstatus.protocol.connection.Connection.remaining") as mock_remaining:
            mock_remaining.side_effect = [15, 208]
```

Reply:
I was thinking while working on this that it was a bit weird, and again I'm not super familiar with writing tests, so thanks for suggestion.